instrumentation/mybatis/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/mybatis/MybatisTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.mybatis;
+
+import static org.mockito.Mockito.when;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import org.apache.ibatis.binding.MapperMethod;
+import org.apache.ibatis.mapping.MappedStatement;
+import org.apache.ibatis.mapping.SqlCommandType;
+import org.apache.ibatis.reflection.factory.DefaultObjectFactory;
+import org.apache.ibatis.session.Configuration;
+import org.apache.ibatis.session.defaults.DefaultSqlSession;
+import org.junit.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.Mockito;
+
+class MybatisTest {
```

Comment:
In my opinion it would be better to use actual mybatis code for testing. Could run an embedded hsqldb or h2 and use mybatis to query that db like we do for hibernate tests. From this test isn't really obvious what this instrumentation really does.

Reply:
I modified the test case based on H2.

Reply:
> In my opinion it would be better to use actual mybatis code for testing. Could run an embedded hsqldb or h2 and use mybatis to query that db like we do for hibernate tests. From this test isn't really obvious what this instrumentation really does.

I modified the test case based on H2.